pkg/metastore/ms_client_test.go

```diff
@@ -64,6 +66,67 @@ func getLocation(repoLocation, branch, tableDir string) string {
 	return fmt.Sprintf("%s/%s/%s", repoLocation, branch, tableDir)
 }
 
+func TestMSClient_CopySchema(t *testing.T) {
+	tests := []struct {
+		TestName         string
+		SourceName       string
+		DestinationName  string
+		LocationURI      string
+		destBranch       string
+		ExpectedLocation string
+		ExpectedError    error
+	}{{
+		TestName:         "with location",
+		SourceName:       "source",
+		DestinationName:  "destination",
+		destBranch:       "branch1",
+		LocationURI:      "s3://example/main/path/to/schema/",
+		ExpectedLocation: "s3://example/branch1/path/to/schema/",
+	}, {
+		TestName:         "table exists",
+		SourceName:       "source",
+		DestinationName:  "source",
+		LocationURI:      "s3://example/path/to/schema/",
+		destBranch:       "branch1",
+		ExpectedLocation: "s3://example/branch1/path/to/schema/",
+		ExpectedError:    msErrors.ErrSchemaExists,
+	},
+		{
+			TestName:         "no location",
+			SourceName:       "source",
+			DestinationName:  "destination",
+			destBranch:       "branch1",
+			LocationURI:      "",
+			ExpectedLocation: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.TestName, func(t *testing.T) {
+			initialDatabases := make(map[string]*metastore.Database)
+
+			initialDatabases[tt.SourceName] = &metastore.Database{Name: tt.SourceName, LocationURI: tt.LocationURI}
+			client := mock.NewMSClient(t, initialDatabases, nil, nil)
+
+			err := metastore.CopyDB(nil, client, client, tt.SourceName, tt.DestinationName, tt.destBranch, "")
+
+			if err != tt.ExpectedError {
+				t.Fatalf("expected err:%s got:%s", tt.ExpectedError, err)
```

Comment:
```suggestion
				t.Fatalf("expected err:%s got:%v", tt.ExpectedError, err)
```
just because the second one can be nil